Datadog.Trace.proj

```diff
@@ -80,8 +80,7 @@
     </MSBuild>
   </Target>
 
-  <Target Name="RestoreAndBuildSamplesForPackageVersions" Condition="'$(TestAllPackageVersions)'=='true'">
-
+  <Target Name="RestoreAndBuildCsharpProjects">
```

Comment:
Urgh, git sucks for tracking this stuff - it looks like you've split the  `RestoreAndBuildSamplesForPackageVersions` target into two, but then you're calling them both twice anyway in the build scripts? Which AFAICT, will result in `RestoreAndBuildCsharpProjects`, `RestoreAndBuildCsharpProjects`, and `RestoreSamplesForPackageVersions` all being called twice? That seems unnecessary? 🤔

Reply:
Well I thought the same... but @zacharycmontoya helped me with this one...

His commit message: 

>The issue with the XUnit test was that it relied on importing the file xunit.runner.visualstudio/<version>/build/<TFM>/xunit.runner.visualstudio.props (I'm guessing to place xunit.runner.visualstudio.dotnetcore.testadapter.dll in the output directory), but it was not imported because the Import evaluation for the Publish MSBuild Task occurred before the Restore MSBuild Task finished. The unoptimized solution for now is to invoke MSBuild twice. This requires two separate MSBuild processes to start up and might cause accidental rebuilds, but it works for now.

I'm open to a better approach for this, but the problem is publishing the XUnit test suite doesn't publish correctly the test adapter, so trying to run the tests will fail.

Reply:
Yikes, what a mess. This may well explain weird behaviours I've seen in other places. That's what we get for trying to use MSBuild to do fancy stuff 🙁 I'll have to work out how this is going to play with the Nuke branch at some point...